evolver/app/tests/test_app.py

```diff
@@ -61,11 +65,31 @@ def test_evolver_app_control_loop_setup(self, app_client):
         response = app_client.get("/")
         assert response.status_code == 200
 
-    def test_schema_endpoint(self, app_client):
-        response = app_client.get("/schema")
+    @pytest.mark.parametrize(
+        "classinfo",
+        (
+            None,
+            Evolver,
+            BaseConfig,
+            BaseInterface,
+            EffectorDriver,
+            SensorDriver,
+            NoOpCalibrator,
+            NoOpEffectorDriver,
+            NoOpSensorDriver,
+        ),
+    )
+    def test_schema_endpoint(self, app_client, classinfo):
+        fqn = evolver.util.fully_qualified_name(classinfo) if classinfo else evolver.util.fully_qualified_name(Evolver)
+        response = app_client.get("/schema", params=dict(classinfo=fqn) if classinfo else None)
         assert response.status_code == 200
         # There's not much in the default config yet, this will change in future PRs.
-        assert json.loads(response.content) == dict(hardware=[], controllers=[])
+        assert json.loads(response.content) == SchemaResponse(classinfo=fqn).model_dump(mode="json")
```

Comment:
might be a good anchor to add a test to pass a simple case through `SchemaResponse` and expect the actual output (or at least significant parts of it). This test would succeed even if SchemaResponse returned `{}` always.